canonicalwebteam/discourse_docs/models.py

```diff
@@ -0,0 +1,302 @@
+        # the "NOTE TO EDITORS" text
+        container = text.parent.parent.parent.parent
+
+        if container.name == "aside" and "quote" in container.attrs["class"]:
+            container.decompose()
+
+    return str(soup)
+
+
+def _replace_notifications(soup):
+    """
+    Given some BeautifulSoup of a document,
+    replace blockquotes with the appropriate notification markup
+    """
+    notification_html = (
+        "<div class='{{ notification_class }}'>"
+        "<div class='p-notification__response'>"
+        "{{ contents | safe }}"
+        "</div></div>"
+    )
+
+    notification_template = Template(notification_html)
+    for note_string in soup.findAll(text=re.compile("ⓘ ")):
+        first_paragraph = note_string.parent
+        blockquote = first_paragraph.parent
+        last_paragraph = blockquote.findChildren(recursive=False)[-1]
+
+        if first_paragraph.name == "p" and blockquote.name == "blockquote":
+            # Remove extra padding/margin
+            first_paragraph.attrs["class"] = "u-no-padding--top"
+            if last_paragraph.name == "p":
+                if "class" in last_paragraph.attrs:
+                    last_paragraph.attrs["class"] += " u-no-margin--bottom"
+                else:
+                    last_paragraph.attrs["class"] = "u-no-margin--bottom"
+
+            # Remove control emoji
+            notification_html = blockquote.encode_contents().decode("utf-8")
+            notification_html = re.sub(
+                r"^\n<p([^>]*)>ⓘ +", r"<p\1>", notification_html
+            )
+
+            notification = notification_template.render(
+                notification_class="p-notification", contents=notification_html
+            )
+            blockquote.replace_with(
+                BeautifulSoup(notification, features="html.parser")
+            )
+
+    for warning in soup.findAll("img", title=":warning:"):
+        first_paragraph = warning.parent
+        blockquote = first_paragraph.parent
+        last_paragraph = blockquote.findChildren(recursive=False)[-1]
+
+        if first_paragraph.name == "p" and blockquote.name == "blockquote":
+            warning.decompose()
+
+            # Remove extra padding/margin
+            first_paragraph.attrs["class"] = "u-no-padding--top"
+            if last_paragraph.name == "p":
+                if "class" in last_paragraph.attrs:
+                    last_paragraph.attrs["class"] += " u-no-margin--bottom"
+                else:
+                    last_paragraph.attrs["class"] = "u-no-margin--bottom"
+
+            notification = notification_template.render(
+                notification_class="p-notification--caution",
+                contents=blockquote.encode_contents().decode("utf-8"),
+            )
+            blockquote.replace_with(
+                BeautifulSoup(notification, features="html.parser")
+            )
+
+    return soup
+
+
+class DiscourseApi:
```

Comment:
Small point, but I would prefer this be `DiscourseAPI`? `API` is [an acronym, and so is capitalised](https://en.wikipedia.org/wiki/Application_programming_interface) - and is [usually also capitalized in package names](https://cwiki.apache.org/confluence/display/CLOUDSTACK/Simple+class+for+making+API+calls%2C+Python).